package/kedro_viz/integrations/deployment/s3_deployer.py

```diff
@@ -0,0 +1,71 @@
+"""`kedro_viz.integrations.deployment.s3_deployer` defines
+deployment class for S3"""
+
+import json
+import logging
+from datetime import datetime
+from pathlib import Path
+
+import fsspec
+from kedro.io.core import get_protocol_and_path
+from semver import VersionInfo
+
+from kedro_viz import __version__
+from kedro_viz.api.rest.responses import save_api_responses_to_fs
+
+_HTML_DIR = Path(__file__).parent.parent.parent.absolute() / "html"
+_METADATA_PATH = "api/deploy-viz-metadata"
+
+logger = logging.getLogger(__name__)
+
+
+class S3Deployer:
+    """Deployer class for AWS S3"""
+
+    def __init__(self, region, bucket_name):
+        self._region = region
+        self._bucket_name = bucket_name
+        self._protocol, self._path = get_protocol_and_path(bucket_name)
+        self._remote_fs = fsspec.filesystem(self._protocol)
+
+    def _upload_api_responses(self):
+        save_api_responses_to_fs(self._bucket_name)
+
+    def _upload_static_files(self):
+        logger.debug("""Uploading static html files to %s.""", self._bucket_name)
+        try:
+            self._remote_fs.put(
+                f"{str(_HTML_DIR)}/*", self._bucket_name, recursive=True
```

Comment:
Instead of hardcoding this `global` value here, I'd add an argument for `html_dir` and then when you call the method do `_upload_static_files(_HTML_DIR)` That way upload static files is a more generic method and can be called with any HTML path in the future.